drivers/temperature/max31855/iio_max31855.c

```diff
@@ -0,0 +1,344 @@
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - 0 in case of success, errno errors otherwise
+*/
+static int max31855_iio_read_raw(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel, intptr_t priv)
+{
+	int ret;
+	struct max31855_iio_dev *max31855_iio;
+	struct max31855_dev *max31855_dev;
+	uint32_t buff;
+	uint32_t temp;
+
+	if (!dev)
+		return -EINVAL;
+
+	max31855_iio = dev;
+	max31855_dev = max31855_iio->max31855_desc;
+
+	switch (channel->type) {
+	case IIO_TEMP:
+		ret = max31855_read_raw(max31855_dev, &buff);
+		if (ret)
+			return ret;
+
+		switch(channel->address) {
+		case 0:
+			temp = NO_OS_GENMASK(15, 0) & buff;
+			temp = no_os_sign_extend32(temp >> 4, 11);
+			break;
+		case 2:
+			temp = (NO_OS_GENMASK(31, 16) & buff) >> 16;
+			temp = no_os_sign_extend32(temp >> 2, 13);
+			break;
+		default:
+			return -EINVAL;
+		}
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &temp);
+
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Handles the read request for scale attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - 0 in case of success, errno errors otherwise
+*/
+static int max31855_iio_read_scale(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel, intptr_t priv)
+{
+	int32_t val[2];
+
+	switch (channel->type) {
+	case IIO_TEMP:
+		switch (channel->address) {
+		case 0:
+			val[0] = 62;
+			val[1] = 500000;
+			return iio_format_value(buf, len, IIO_VAL_INT_PLUS_MICRO, 2, val);
+		case 2:
+			val[0] = 250;
+			return iio_format_value(buf, len, IIO_VAL_INT, 1, val);
+		default:
+			return -EINVAL;
+		}
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Reads the number of given samples for the selected channels
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param samples - Number of samples to be returned
+ * @return ret    - 0 in case of success, errno errors otherwise
+*/
+static int max31855_iio_read_samples(void* dev, uint16_t* buff,
+				     uint32_t samples)
+{
+	int ret;
+	uint32_t i_temp;
+	uint32_t t_temp;
+	uint32_t raw_buff;
+	struct max31855_iio_dev *max31855_iio;
+	struct max31855_dev *max31855_dev;
+
+	if (!dev)
+		return -EINVAL;
+
+	max31855_iio = dev;
+	max31855_dev = max31855_iio->max31855_desc;
+
+	for (uint32_t i = 0; i < samples;) {
+		ret = max31855_read_raw(max31855_dev, &raw_buff);
+		if (ret)
+			return ret;
+
+		if (max31855_iio->active_channels & NO_OS_BIT(0)) {
+			i_temp = NO_OS_GENMASK(15, 0) & raw_buff;
+			i_temp = no_os_sign_extend32(i_temp, 15);
+			buff[i] = i_temp;
```

Comment:
so my assumption is that scan_index 0 corresponds to b0 of active_channels
scan_index 1 corresponds to  b1 of active_channels

is it possible that in this case i_temp is shown when t_temp is selected and vice-versa?